nomt/src/merkle/page_walker.rs

```diff
@@ -54,10 +54,17 @@ use nomt_core::{
 
 use crate::{
     merkle::BucketInfo,
+    metrics::Metrics,
     page_cache::{Page, PageMut},
     page_diff::PageDiff,
 };
 
+/// Threshold representing the number of leaves required to be present in the two
+/// subtrees contained in a page to be stored on disk.
+/// If this threshold is not exceeded, the page will not be stored on disk
+/// and will be constructed on the fly when needed.
+const PAGE_ELISION_THRESHOLD: u64 = 10;
```

Comment:
I would prefer if this is tunable. Also, I think that the value here can be higher without impacting perf.